pkg/config/nerdctl_config_applier.go

```diff
@@ -96,6 +100,12 @@ func updateNerdctlConfig(fs afero.Fs, user string, rootless bool) error {
 	}
 
 	var cfg Nerdctl
+	if (fc.HostGatewayIP == nil) {
```

Comment:
@TrungBui59 I just made it work by moving the logic after line 116.

Basically line 109-116 reads the current existing nerdctl config and fill it to `cfg`. So in vm start scenario, the existing host gateway ip will be filled to `cfg` and overwrites the value you filled in line 106.

Moving it after 116 could let you overwrite the value of existing nerdctl config.